File: src/main/java/com/sparetimedevs/questlog/RootResourceProcessor.java
package com.sparetimedevs.questlog;

import com.sparetimedevs.questlog.login.LoginController;
import org.springframework.data.rest.webmvc.RepositoryLinksResource;
import org.springframework.hateoas.ResourceProcessor;
import org.springframework.stereotype.Component;

import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;

@Component
public class RootResourceProcessor implements ResourceProcessor<RepositoryLinksResource> {

	@Override
	public RepositoryLinksResource process(RepositoryLinksResource resource) {
		resource.add(linkTo(LoginController.class).withRel("login"));
		return resource;
	}
}

File: src/main/java/com/sparetimedevs/weeklyachievements/RootResourceProcessor.kt
package com.sparetimedevs.weeklyachievements

import com.sparetimedevs.weeklyachievements.login.LoginController
import org.springframework.data.rest.webmvc.RepositoryLinksResource
import org.springframework.hateoas.ResourceProcessor
import org.springframework.stereotype.Component

import org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo

@Component
class RootResourceProcessor : ResourceProcessor<RepositoryLinksResource> {

	override fun process(resource: RepositoryLinksResource): RepositoryLinksResource {
		resource.add(linkTo(LoginController::class.java).withRel("login"))
		return resource
	}
}
